tests/test_hubmixin.py

```diff
@@ -133,3 +134,25 @@ def test_push_to_hub(self):
         self.assertEqual(model_info.modelId, f"{USER}/{REPO_NAME}")
 
         self._api.delete_repo(repo_id=f"{REPO_NAME}", token=self._token)
+
+    def test_auto_lfs(self):
+
+        REPO_NAME = repo_name("PUSH_TO_HUB")
+        model = DummyModel()
+        large_file = [100] * int(4e6)
+        model.save_pretrained(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        with open(f"{WORKING_REPO_DIR}/{REPO_NAME}/large_file.txt", "w+") as f:
+            f.write(json.dumps(large_file))
```

Comment:
Why are you saving in line 143? `push_to_hub` already calls `save_pretrained` at https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/hub_mixin.py#L319

Reply:
It creates a repository to which I can save a large file in. If I first save the big file, I will not be able to open a repository inside target directory and push through push to hub, because it saves to a non-empty folder.

Reply:
mmm `save_pretrained` does not create a `git` repo, it will just save the model in the directory. `push_to_hub` will automatically create the repo and save the files there, as seen in https://github.com/huggingface/huggingface_hub/blob/f7627b412e7a9f778cf382ce934f4a5f7b5e4409/tests/test_hubmixin.py#L120-L130

Reply:
If the issue is because it's not a valid git repository, why not make it one? E.g. you can use `Repository()` to create a file before creating the large file.
(note this might end up being deprecated before next release so not sure if it's still worth fixing though, but the issue you mention should be solvable just creating a repo)